lib/Parse/ParseDecl.cpp

```diff
@@ -1460,6 +1347,41 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
         .fixItReplace(Tok.getLoc(), "available");
   }
 
+
+  if (DK == DAK_Count && Tok.getText() == "warn_unused_result") {
```

Comment:
Looking at this more, I'm not sure that having a new attribute macro for removed attributes is going to help much. It will still require having the actual token somewhere in code and someplace to insert a fixit (like this block does now); however, in addition, it will require checks in most of the places where I touched, some of which will still need to use the `DAK_<AttrName>`.

One of the nice parts of doing it this way is that I could search/grep for "WarnUnusedResult` and find all the places I needed to change. After the change, the only places with that string are this block and tests for this block.